src/peft/utils/safelora.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2024-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Reference paper: https://arxiv.org/abs/2405.16833
+
+
+import copy
+import os
+from dataclasses import dataclass, field
+from typing import Literal
+
+import torch
+from safetensors import safe_open
+from safetensors.torch import save_file
+
+from peft import PeftConfig
+
+from .loftq_utils import _SafetensorLoader as SafetensorLoader
+
+
+@dataclass
+class SafeLoraConfig:
+    """
+    This is the configuration class to store the configuration of a safeLora.
+
+
+    Args:
+
+    base_model_path (`str`): The path of the base model for obtaining the aligned matrix.
+
+    aligned_model_path (`str`): The path of the aligned model for obtaining the aligned matrix.
+
+    peft_model_path (`str`): The path of the LoRA wieghts and configs.
+
+    select_layers_type (`Literal["threshold", "number"]`): How to select projection layers? options: [threshold, number]
+
+    threshold (`float`): The threshold of cosine similarity for selecting projected layers.
+
+    num_proj_layers (`int`): The number of projected layers.
+
+    device (`str`): Device is used in SafeLoRA (cuda or cpu).
+
+    save_weights (`bool`): Replacing and saving SafeLoRA weights to the original LoRA file.
+
+    local_files_only (`bool`): Using for snapshot_download.
+
+    dtype (`torch.dtype`): Data type for model weights, e.g., torch.float32 or torch.bfloat16.
+
+    """
+
+    base_model_path: str = field(
+        default="meta-llama/Llama-2-7b-hf",
+        metadata={"help": "The path of the base model for obtaining the aligned matrix."},
+    )
+
+    aligned_model_path: str = field(
+        default="TheBloke/Llama-2-7B-Chat-fp16",
+        metadata={"help": "The path of the aligned model for obtaining the aligned matrix."},
+    )
+
+    peft_model_path: str = field(
+        default="LisaSchunke/llama-2-7b-peft-finetuned-20000-dataset",
```

Comment:
IMO, it doesn't make sense to set default values here, I would remove them. WDYT?